### DISCUSSION TITLE
Variables: Add timezone as global template variable

### DISCUSSION THREAD

**Discussion by DrMichael** [grafana/grafana#38159] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
I would like to have the timezone in which the data is displayed available in queries. E.g. when using date_trunc('day', ...) in psql, it gets truncated to midnight of the database server timezone, but displayed in the display timezone.

**Why is this needed**:
To correct above mentioned behaviour one needs the display timezone available in queries.
Or to be able to set the psql client timezone.

**Comment by VladSnap** [comment]
Hello everyone. It would be just as helpful for me. 
+I would like the global variables to be well worked out in general.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by j0j0i996** [comment]
I would like to request this feature as well, but have a different use case: 

I am dealing with forecast timeseries data with 2 time columns. One timestamp represents the time for which the forecast is made ('forecast_time') and the other represents the time the forecast was created ('issued_time'). In Grafana, I use 'forecast_time' as base for the regular timeseries and would like to expose 'issued_time' in the name of the field. To show the correct 'issued_timestamp', I need to be able to dynamically shift it to the dashboard timezone using a global variable.

So far I can only hardcode my dashboard timezone: 
`issued_timezone at time zone 'utc' at time zone 'Europe/Berlin'`

**Reply by cschlipf** [reply]
Instead of hardcoding it right into the query I would recommend adding a timezone dashboard variable that your can use in your query. Still not ideal as you have to set this manually on every single dashboard, when it would be better to use the logged in User's timezone from the Grafana setting. But much better than hardcoding it into every single query.

**Comment by dnnshssmnn** [comment]
I have a very similar usecase like @DrMichael for this feature (although using BigQuery as datasource).

In addition to that, I also need the selected timezone to use bar charts properly: for example I have to group by hour on the server side where everything is stored in UTC and therefore it will be displayed in UTC. With the global variable timezone I could adjust the results according to the selcted timezone in the Grafana dashboard and display the desired results correctly.

I would greatly appreciate if this feature would make it into some near future release.

In addition, I want to point out others that have asked for this feature in the Grafana forum as well:
https://community.grafana.com/t/fetch-current-time-zone-through-a-variable/53225
https://community.grafana.com/t/no-timezone-global-variable/66172

**Reply by mehtakartik** [reply]
My scenario:
The postgres DB timezone is India. Data is inserted in a timestamp column (no timezone information).
The viewers of the grafana dashboard are in PST and IST. Both teams need to interpret that time in PST. However, Grafana assumes data is stored in UTC, and unless the timezone of the timepicker is in UTC,  the time is shown incorrectly in both timezones. To correct that, I need to offset the time in my query in an opposite direction to where grafana is being viewed, to always show the same time irrespective of the selected timezone. I need a timezone variable for that, so that I can use it in my query to postgres.

**Comment by ahirner** [comment]
I found a workaround to inject the user's TZ offset for postgres. 

```sql
-- should return HH:MM from user's browser if that's also the dashboard's TZ setting
select * from to_char(now() + interval '${__from:date:Z}', 'HH24:MI');
```

It probably applies to all datasources that can parse and add [Z or ZZ](https://momentjs.com/docs/#/displaying/format/) formatted date parts.

**Comment by VictorColomb** [comment]
You'll have to wait for v10.1.0 but #66785 😉
